### ISSUE TITLE
Turbine Test: Win Liberty Project Create Failures

### ISSUE THREAD

**Issue by maysunfaisal** [eclipse/codewind#1062]
**Codewind version:** latest
**OS:** Win

**Che version:**
**IDE extension version:**
**IDE version:**
**Kubernetes cluster:**

**Description:**
The Win Turbine func tests seem to be failing on Liberty create
```

[34m[08/11/19 06:01:27 codewindtestliberty] [TRACE][39m  ---> 1adafcb9c97d

[34m[08/11/19 06:01:27 codewindtestliberty] [TRACE][39m Step 24/24 : USER default

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m  ---> Running in fe249875871d

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m Removing intermediate container fe249875871d

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m  ---> 4f239e7ed4ac

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m Successfully built 4f239e7ed4ac

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m Successfully tagged cw-codewindtestliberty-liberty1573190740157:latest

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m Successfully built cw-codewindtestliberty-liberty1573190740157

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m Container image has been built successfully
Updating build state for project codewindtestliberty(liberty1573190740157) to: inProgress

[91m[08/11/19 06:01:28 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | Update build state request failed for project: codewindtestliberty(liberty1573190740157), with code: 404, and message: Not Found

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m Container built
* Starting container. 

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m Start the container
Run command to start the container: /file-watcher/idc/artifacts/run_docker.sh cw-codewindtestliberty-liberty1573190740157 cw-codewindtestliberty-liberty1573190740157 "-p 127.0.0.1::7777 -P" /codewind-workspace/.idc/docker/docker-c6f926dbd9ee2b9006ad19e42c84133e14bfafec /codewind-workspace "codewindtestliberty-liberty1573190740157"
[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m 

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m Running codewind app container for /codewind-workspace/codewindtestliberty using container name cw-codewindtestliberty-liberty1573190740157

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m Application path used for volume mounting is: /codewind-workspace/codewindtestliberty

[34m[08/11/19 06:01:28 codewindtestliberty] [TRACE][39m Log path used for volume mounting is: /codewind-workspace/.logs/codewindtestliberty-liberty1573190740157

[36m[08/11/19 06:01:28 Turbine] [DEBUG][39m Running Builds queue: ["liberty1573190740157"]
[36m[08/11/19 06:01:28 Turbine] [DEBUG][39m Builds in progress: 1
[34m[08/11/19 06:01:31 codewindtestliberty] [TRACE][39m Copying over source files

[36m[08/11/19 06:01:33 Turbine] [DEBUG][39m Running Builds queue: ["liberty1573190740157"]
[36m[08/11/19 06:01:33 Turbine] [DEBUG][39m Builds in progress: 1
[91m[08/11/19 06:01:34 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | Error response from daemon: Error processing tar file(exit status 1): lchown /home/default/app/libertyrepocache.zip: no such file or directory

[34m[08/11/19 06:01:34 codewindtestliberty] [TRACE][39m 81329401b02c9a67258c5acf13df4b567e54bc56df99a4e1eea95e61a84733c9

[34m[08/11/19 06:01:34 codewindtestliberty] [TRACE][39m Stored application container for codewindtestliberty: 81329401b02c9a67258c5acf13df4b567e54bc56df99a4e1eea95e61a84733c9

[32m[08/11/19 06:01:34 codewindtestliberty] [INFO][39m Internal port: 9080
[32m[08/11/19 06:01:34 codewindtestliberty] [INFO][39m Exposed port: 32784
[32m[08/11/19 06:01:34 codewindtestliberty] [INFO][39m Internal debug port: 
[32m[08/11/19 06:01:34 codewindtestliberty] [INFO][39m Exposed debug port: 
[32m[08/11/19 06:01:34 codewindtestliberty] [INFO][39m All the exposed container ports: ["7777","9080","9443"]
[32m[08/11/19 06:01:34 codewindtestliberty] [INFO][39m All the host ports: ["32785","32784","32783"]
[34m[08/11/19 06:01:34 codewindtestliberty] [TRACE][39m Log file name for build: cw-liberty1573190740157-c6f926dbd9ee2b9006ad19e42c84133e14bfafec

[34m[08/11/19 06:01:34 codewindtestliberty] [TRACE][39m Compiling application for project codewindtestliberty
Updating build state for project codewindtestliberty(liberty1573190740157) to: inProgress

[34m[08/11/19 06:01:34 codewindtestliberty] [TRACE][39m Touching maven build log file: /home/default/logs/maven.build.log

[91m[08/11/19 06:01:34 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | Update build state request failed for project: codewindtestliberty(liberty1573190740157), with code: 404, and message: Not Found

[34m[08/11/19 06:01:34 codewindtestliberty] [TRACE][39m Extracting maven m2 cache to /home/default

[91m[08/11/19 06:01:35 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | java.io.FileNotFoundException: /home/default/app/localm2cache.zip (No such file or directory)

[91m[08/11/19 06:01:35 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | 	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:212)

[91m[08/11/19 06:01:35 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | 	at java.io.FileInputStream.<init>(FileInputStream.java:152)

[91m[08/11/19 06:01:35 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | 	at java.io.FileInputStream.<init>(FileInputStream.java:104)

[91m[08/11/19 06:01:35 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | 	at sun.tools.jar.Main.run(Main.java:319)

[91m[08/11/19 06:01:35 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | 	at sun.tools.jar.Main.main(Main.java:1300)

[34m[08/11/19 06:01:35 codewindtestliberty] [TRACE][39m Finished extracting maven m2 cache to /home/default

[34m[08/11/19 06:01:35 codewindtestliberty] [TRACE][39m Maven m2 cache is not set up for cw-liberty1573190740157-c6f926dbd9ee2b9006ad19e42c84133e14bfafec
Maven build output directory is set to /tmp/liberty

[34m[08/11/19 06:01:35 codewindtestliberty] [TRACE][39m Start mvn package for cw-liberty1573190740157-c6f926dbd9ee2b9006ad19e42c84133e14bfafec Fri Nov  8 06:01:35 UTC 2019
mvn -B package -DskipTests=true -DlibertyEnv=microclimate -DmicroclimateOutputDir=/tmp/liberty  --log-file /home/default/logs/maven.build.log

[34m[08/11/19 06:01:38 codewindtestliberty] [TRACE][39m Finished mvn package for cw-liberty1573190740157-c6f926dbd9ee2b9006ad19e42c84133e14bfafec Fri Nov  8 06:01:38 UTC 2019

[34m[08/11/19 06:01:38 codewindtestliberty] [TRACE][39m Triggering log file event for: maven build log

[91m[08/11/19 06:01:38 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | Update app state request failed for project: codewindtestliberty(liberty1573190740157), with code: 404, and message: Not Found

[36m[08/11/19 06:01:38 Turbine] [DEBUG][39m Running Builds queue: ["liberty1573190740157"]
[36m[08/11/19 06:01:38 Turbine] [DEBUG][39m Builds in progress: 1
[34m[08/11/19 06:01:39 codewindtestliberty] [TRACE][39m [INFO] Scanning for projects...

[34m[08/11/19 06:01:39 codewindtestliberty] [TRACE][39m [INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.109 s
[INFO] Finished at: 2019-11-08T06:01:38Z

[34m[08/11/19 06:01:39 codewindtestliberty] [TRACE][39m [INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/home/default/app). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 

[34m[08/11/19 06:01:39 codewindtestliberty] [TRACE][39m [ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException

[91m[08/11/19 06:01:39 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | Application build failed for project codewindtestliberty. Please check the log in the build log view for more details

[34m[08/11/19 06:01:39 codewindtestliberty] [TRACE][39m Updating build state for project codewindtestliberty(liberty1573190740157) to: failed

[91m[08/11/19 06:01:39 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | Update build state request failed for project: codewindtestliberty(liberty1573190740157), with code: 404, and message: Not Found

[91m[08/11/19 06:01:39 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | Application build failed.

[91m[08/11/19 06:01:39 codewindtestliberty] [ERROR][39m [31m[ File Name: /file-watcher/server/src/projects/projectUtil.ts | Function Name: processManager.spawnDetached | Line Number: 48] [39mError code: 3 - The container failed to start for application /codewind-workspace/codewindtestliberty
[34m[08/11/19 06:01:39 Turbine] [TRACE][39m Key translated. Returning message: Creating image for build failed
[32m[08/11/19 06:01:39 codewindtestliberty] [INFO][39m Validating project using operationId: cd16343587d71b9951f166f3b3fc998f
[34m[08/11/19 06:01:39 Turbine] [TRACE][39m {"projectType":"liberty","location":"/codewind-workspace/codewindtestliberty","projectID":"liberty1573190740157"}
[32m[08/11/19 06:01:39 codewindtestliberty] [INFO][39m Checking for required files
[32m[08/11/19 06:01:39 codewindtestliberty] [INFO][39m Checking for required file /Dockerfile | /Dockerfile-lang
[32m[08/11/19 06:01:39 codewindtestliberty] [INFO][39m Checking for required file /Dockerfile-build
[32m[08/11/19 06:01:39 codewindtestliberty] [INFO][39m Checking for required file /pom.xml
[34m[08/11/19 06:01:39 codewindtestliberty] [TRACE][39m idc build failed for /codewind-workspace/codewindtestliberty with exit code 6 Fri Nov  8 06:01:39 UTC 2019

[91m[08/11/19 06:01:39 codewindtestliberty] [ERROR][39m executeBuildScript : 48 | | The command /file-watcher/scripts/liberty-container.sh /codewind-workspace/codewindtestliberty,/codewind-workspace,liberty1573190740157,create,cw-codewindtestliberty-liberty1573190740157,true,cw-liberty1573190740157-c6f926dbd9ee2b9006ad19e42c84133e14bfafec,run,7777,NONE,codewindtestliberty-liberty1573190740157,, failed with exit code 3
[32m[08/11/19 06:01:39 codewindtestliberty] [INFO][39m Build state changed for project: liberty1573190740157 from: inProgress, to: failed, with message: buildscripts.buildFail
[32m[08/11/19 06:01:39 codewindtestliberty] [INFO][39m Emitting event 
 message: projectStatusChanged
 data: {
  "projectID": "liberty1573190740157",
  "buildStatus": "failed",
  "detailedBuildStatus": "Creating image for build failed",
  "lastbuild": 1573192899250
}
```

Looks like this line https://github.com/eclipse/codewind/blob/master/src/pfe/file-watcher/idc/artifacts/run_docker.sh#L67 is failing and it does not copy over the entire src and mvn fails cuz there is no pom.xml

But we are able to manually create Liberty projects on Win without any issue(as of Nov 7 5pm EST)

**Comment by ssh24** [comment]
@rajivnathan Please assign this to me.

**Comment by ssh24** [comment]
@maysunfaisal Could you please verify and close this?

**Comment by maysunfaisal** [comment]
Seeing this today

**Comment by ssh24** [comment]
related to #1549

**Comment by maysunfaisal** [comment]
During Microprofile create, creation fails due to run_docker.sh failing on docker cp of project root to the app container.

It fails on https://github.com/ssh24/codewind/blob/a0057bdb39f17ed9bf5aaedbfba53d41198851d2/src/pfe/file-watcher/idc/artifacts/run_docker.sh#L70

@ssh24 debugged this and found this in the log before docker cp, docker exit code, after docker cp:

```
[06/12/19 18:33:35 turbinetest-default-liberty] [TRACE] Listing source files on PFE container

[06/12/19 18:33:35 turbinetest-default-liberty] [TRACE] total 220596
drwxrwxr-x  5 1001 root      4096 Dec  6 15:59 .
drwxrwxrwx 26 root root      4096 Nov 27 02:12 ..
-rwxrwxr-x  1 1001 root       203 Dec  6 18:25 .cw-settings
-rwxrwxr-x  1 1001 root       607 Dec  6 18:25 Dockerfile
-rwxrwxr-x  1 1001 root      2664 Dec  6 18:25 Dockerfile-build
-rwxrwxr-x  1 1001 root       366 Dec  6 18:25 Dockerfile-tools
-rwxrwxr-x  1 1001 root       116 Dec  6 18:25 Jenkinsfile
-rwxrwxr-x  1 1001 root      2915 Dec  6 18:25 README.md
drwxrwxr-x  3 1001 root      4096 Dec  6 15:59 chart
-rw-rwxr--  1 1001 root  61314717 Jun 18 20:58 libertyrepocache.zip
-rw-rwxr--  1 1001 root 164499767 Aug 28 19:56 localm2cache.zip
-rwxrwxr-x  1 1001 root     22882 Dec  6 18:25 pom.xml
drwxrwxr-x  4 1001 root      4096 Dec  6 15:59 src
drwxrwxr-x  3 1001 root      4096 Nov 27 02:13 target

[06/12/19 18:33:35 turbinetest-default-liberty] [TRACE] Copying over source files

[06/12/19 18:33:39 Turbine] [DEBUG] Running Builds queue: ["default-liberty-1575656847622"]
[06/12/19 18:33:39 Turbine] [DEBUG] Builds in progress: 1
[06/12/19 18:33:44 Turbine] [DEBUG] Running Builds queue: ["default-liberty-1575656847622"]
[06/12/19 18:33:44 Turbine] [DEBUG] Builds in progress: 1
[06/12/19 18:33:49 Turbine] [DEBUG] Running Builds queue: ["default-liberty-1575656847622"]
[06/12/19 18:33:49 Turbine] [DEBUG] Builds in progress: 1
[06/12/19 18:33:54 turbinetest-default-liberty] [ERROR] executeBuildScript : 48 | Error response from daemon: Error processing tar file(exit status 1): lchown /home/default/app/libertyrepocache.zip: no such file or directory

[06/12/19 18:33:54 turbinetest-default-liberty] [TRACE] Exit code of docker cp command: 1
Listing source files on APP container

[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] Internal port: 9080
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] Exposed port: 33330
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] Internal debug port:
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] Exposed debug port:
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] All the exposed container ports: ["7777","9080","9443"]
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] All the host ports: ["33331","33330","33329"]
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] Internal port: 9080
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] Exposed port: 33330
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] Internal debug port:
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] Exposed debug port:
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] All the exposed container ports: ["7777","9080","9443"]
[06/12/19 18:33:54 turbinetest-default-liberty] [INFO] All the host ports: ["33331","33330","33329"]
[06/12/19 18:33:55 Turbine] [DEBUG] Running Builds queue: ["default-liberty-1575656847622"]
[06/12/19 18:33:55 Turbine] [DEBUG] Builds in progress: 1
[06/12/19 18:33:55 turbinetest-default-liberty] [TRACE] total 36
drwxrwxr-x 3 default root 4096 Dec  6 18:33 .
drwxrwxr-x 1 default root 4096 Dec  6 18:33 ..
-rwxrwxr-x 1 default root  203 Dec  6 18:25 .cw-settings
-rwxrwxr-x 1 default root  607 Dec  6 18:25 Dockerfile
-rwxrwxr-x 1 default root 2664 Dec  6 18:25 Dockerfile-build
-rwxrwxr-x 1 default root  366 Dec  6 18:25 Dockerfile-tools
-rwxrwxr-x 1 default root  116 Dec  6 18:25 Jenkinsfile
-rwxrwxr-x 1 default root 2915 Dec  6 18:25 README.md
drwxrwxr-x 3 default root 4096 Dec  6 18:33 chart
```

**Comment by maysunfaisal** [comment]
Saw this issue on Win test on Jan 06 and Jan 07 and Linux test on Jan 06

cc @rajivnathan

**Comment by maysunfaisal** [comment]
Docker pull down the cache to an another path other than the app dir and see if it resolves.

if it does, then the entrypoint rm -rf is the issue, since entrypoint might collide with when we're doing docker cp

**Comment by ssh24** [comment]
This was fixed by #1799 and have been passing since then.

/assign maysunfaisal
/close
